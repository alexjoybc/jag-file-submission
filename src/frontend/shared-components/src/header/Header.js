/* eslint-disable no-alert */
import React from "react";
import PropTypes from "prop-types";
import "./Header.css";
import bcidSymbol from "../../public/images/bcid-symbol-rev.svg";
import bcidLogoRev from "../../public/images/bcid-logo-rev-en.svg";

export const HeadingTitle = classNames => (
  <div className={classNames} aria-labelledby="title" />
);

export const HeaderImage = (history, classNames, width, src) => (
  <img
    className={classNames}
    src={src}
    width={width}
    height="44"
    alt="B.C. Government Logo"
    onClick={() => history.push("/")}
    role="button"
    onKeyDown={() => history.push("/")}
    tabIndex={0}
  />
);

export default function Header({ header: { name, history } }) {
  return (
    <header>
      <nav className="container-fluid navbar navbar-expand-lg navbar-dark">
        {HeadingTitle("navbar-brand pointer")}
        {HeaderImage(
          history,
          "img-fluid d-none d-md-block pointer",
          181,
          bcidLogoRev
        )}
        {HeaderImage(history, "img-fluid d-md-none pointer", 64, bcidSymbol)}
        {HeadingTitle("pointer navbar-brand nav-item nav-link")}
        <div id="title" className="navbar-brand">
          {name}
        </div>
      </nav>
    </header>
  );
}

Header.propTypes = {
  header: PropTypes.shape({
    name: PropTypes.string.isRequired,
    history: PropTypes.any.isRequired
  }).isRequired
};
