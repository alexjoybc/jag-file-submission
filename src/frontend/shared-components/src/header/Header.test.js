import React from "react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import { render, fireEvent, getAllByRole } from "@testing-library/react";
import testBasicSnapshot from "../TestHelper";
import Header, { HeadingTitle, HeaderImage } from "./Header";

describe("Header Component", () => {
  const header = {
    name: "File Submission"
  };
  const history = createMemoryHistory();

  const headerComponent = (
    <Router history={history}>
      <Header header={header} />
    </Router>
  );

  test("Header matches the snapshot", () => {
    testBasicSnapshot(headerComponent);
  });

  test("HeadingTitle matches the snapshot", () => {
    const headingTitle = HeadingTitle(history, "navbar-brand pointer");

    testBasicSnapshot(headingTitle);
  });

  test("HeaderImage matches the snapshot", () => {
    const headingImage = HeaderImage(
      "img-fluid d-none d-md-block",
      181,
      "bcid-logo-rev-en.svg"
    );

    testBasicSnapshot(headingImage);
  });

  test("Clicking HeadingTitle takes you back to home", () => {
    history.location.pathname = "/somepageroute";

    const { container } = render(headerComponent);

    fireEvent.click(getAllByRole(container, "button")[0]);

    expect(history.location.pathname).toEqual("/");
  });

  test("Keydown on HeadingTitle takes you back to home", () => {
    history.location.pathname = "/somepageroute";

    const { container } = render(headerComponent);

    fireEvent.keyDown(getAllByRole(container, "button")[0]);

    expect(history.location.pathname).toEqual("/");
  });
});
