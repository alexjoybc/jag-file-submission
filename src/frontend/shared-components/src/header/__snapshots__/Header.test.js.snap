// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Header matches the snapshot 1`] = `
<header>
  <nav
    className="container-fluid navbar navbar-expand-lg navbar-dark"
  >
    <div
      aria-labelledby="title"
      className="navbar-brand pointer"
    />
    <img
      alt="B.C. Government Logo"
      className="img-fluid d-none d-md-block pointer"
      height="44"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
      width={181}
    />
    <img
      alt="B.C. Government Logo"
      className="img-fluid d-md-none pointer"
      height="44"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
      width={64}
    />
    <div
      aria-labelledby="title"
      className="pointer navbar-brand nav-item nav-link"
    />
    <div
      className="navbar-brand"
      id="title"
    >
      File Submission
    </div>
  </nav>
</header>
`;

exports[`Header Component HeaderImage matches the snapshot 1`] = `
<img
  alt="B.C. Government Logo"
  className={181}
  height="44"
  onClick={[Function]}
  width="bcid-logo-rev-en.svg"
/>
`;

exports[`Header Component HeadingTitle matches the snapshot 1`] = `
<div
  aria-labelledby="title"
  className={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "ml5b7a",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "ml5b7a",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
/>
`;
